src/shared/state/assets.ts

```diff
@@ -0,0 +1,63 @@
+import { makeAutoObservable, when } from 'mobx';
+import { ViewService } from '@penumbra-zone/protobuf';
+import { Metadata } from '@penumbra-zone/protobuf/penumbra/core/asset/v1/asset_pb';
+import { getDenomMetadata } from '@penumbra-zone/getters/assets-response';
+import { ChainRegistryClient } from '@penumbra-labs/registry';
+import { Constants } from '@/shared/configConstants';
+import { penumbra } from '@/shared/penumbra';
+import { connectionStore } from '@/shared/state/connection';
+import { queryClient } from '@/shared/queryClient';
+
+const fetchAssets = () => {
+  return queryClient.fetchQuery({
```

Comment:
used react-query to apply caching for the fetched data but not using `useQuery` since there are multiple requests in this store – couldn't figure out how to merge them together